packages/neovim-nightly/build.sh

```diff
@@ -63,9 +63,23 @@ termux_pkg_auto_update() {
 		return 0
 	fi
 
+	# Use current time in the package version, as
+	# 1. The version number without timestrap is lesser than the one with timestrap, a.k.a
+	#    dpkg --compare-version "0.8.0-dev-nightly-9-g1ef84547a" le "0.8.0-dev-nightly-20220725120000-9-g1ef84547a", returns 0
+	# 2. The version number with older timestrap is lesser than the one with newer timestrap, a.k.a
+	#    dpkg --compare-version "0.8.0-dev-nightly-20220725110000-1-g123456789" le "0.8.0-dev-nightly-20220725120000-9-g1ef84547a", returns 0
+	# 3. When auto-updating, we actually ignore the timestrap, just like the version check below.
+	local _current_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$TERMUX_PKG_VERSION")"
+	local _current_nvim_version_suffix="$(grep -oP '\d+-g[0-9a-f]+' <<< "$TERMUX_PKG_VERSION")"
+	local _current_nvim_version="$_current_nvim_version_prefix$_current_nvim_version_suffix"
+	local _remote_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$remote_nvim_version")"
+	local _remote_nvim_version_suffix="$(grep -oP '\-\d+-g[0-9a-f]+' <<< "$remote_nvim_version")"
+	local _current_time=$(date "+%Y%m%d%H%M%S")
+	local _package_version_to_update="$_remote_nvim_version_prefix$_current_time$_remote_nvim_version_suffix"
+
 	# since we are using a nightly build, therefore no need to check for version increment/decrement.
-	if [ "${TERMUX_PKG_VERSION}" != "${remote_nvim_version}" ]; then
-		termux_pkg_upgrade_version "${remote_nvim_version}" --skip-version-check
+	if [ "${_current_nvim_version}" != "${remote_nvim_version}" ]; then
```

Comment:
This will always be true each time the bot check for updates

Reply:
It's not like what you think.

Let us assume that `TERMUX_PKG_VERSION="0.8.0-dev-nightly-202207250000-4-ge9b58a619"`, `remote_nvim_version=0.8.0-dev-nightly-9-g1ef84547a`.
```
_current_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$TERMUX_PKG_VERSION")" # "0.8.0-dev-nightly-"
_current_nvim_version_suffix="$(grep -oP '\d+-g[0-9a-f]+' <<< "$TERMUX_PKG_VERSION")"               # "4-ge9b58a619"
_current_nvim_version="$_current_nvim_version_prefix$_current_nvim_version_suffix"                  # "0.8.0-dev-nightly-4-ge9b58a619"
_remote_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$remote_nvim_version")" # "0.8.0-dev-nightly-"
_remote_nvim_version_suffix="$(grep -oP '\-\d+-g[0-9a-f]+' <<< "$remote_nvim_version")"             # "-9-g1ef84547a"
_current_time=$(date "+%Y%m%d%H%M%S")                                                               # "20220725225350"
_package_version_to_update="$_remote_nvim_version_prefix$_current_time$_remote_nvim_version_suffix" # "0.8.0-dev-nightly-20220725225350-9-g1ef84547a"
[ "${_current_nvim_version}" != "${remote_nvim_version}" ] # [ "0.8.0-dev-nightly-4-ge9b58a619" != "0.8.0-dev-nightly-9-g1ef84547a" ] # true
```

And if `TERMUX_PKG_VERSION="0.8.0-dev-nightly-202207250000-4-ge9b58a619"`,`remote_nvim_version=0.8.0-dev-nightly-4-ge9b58a619`,
```
_current_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$TERMUX_PKG_VERSION")" # "0.8.0-dev-nightly-"
_current_nvim_version_suffix="$(grep -oP '\d+-g[0-9a-f]+' <<< "$TERMUX_PKG_VERSION")"               # "4-ge9b58a619"
_current_nvim_version="$_current_nvim_version_prefix$_current_nvim_version_suffix"                  # "0.8.0-dev-nightly-4-ge9b58a619"
_remote_nvim_version_prefix="$(grep -oP '^\d+\.\d+\.\d+-dev-nightly-*' <<< "$remote_nvim_version")" # "0.8.0-dev-nightly-"
_remote_nvim_version_suffix="$(grep -oP '\-\d+-g[0-9a-f]+' <<< "$remote_nvim_version")"             # "-4-ge9b58a619"
_current_time=$(date "+%Y%m%d%H%M%S")                                                               # "20220725225350"
_package_version_to_update="$_remote_nvim_version_prefix$_current_time$_remote_nvim_version_suffix" # "0.8.0-dev-nightly-20220725225632-4-ge9b58a619"
[ "${_current_nvim_version}" != "${remote_nvim_version}" ] # [ "0.8.0-dev-nightly-4-ge9b58a619" != "0.8.0-dev-nightly-4-ge9b58a619" ] # false
```

Why `[ "${_current_nvim_version}" != "${remote_nvim_version}" ]` will always be true?